### DISCUSSION TITLE
Integrate NestJS in NextJS using http.server?

### DISCUSSION THREAD

**Discussion by arashi-dev** [vercel/next.js#35638] [Help]
hello.

I've recently saw this article [https://simonknott.de/articles/integrating-nextjs-with-nestjs](https://simonknott.de/articles/integrating-nextjs-with-nestjs)
He integrated NestJS and NextJS via `/api` routes and `http.server` of NestJS.

Is it a good approach? is it stable to make API with NestJS with this approach?

**Comment by raphaelbadia** [comment]
Even though I love the concept, I'm not sure nestjs is suited for serverless api calls - it seems to require some optimisations to work well in serverless https://docs.nestjs.com/faq/serverless
